borg/remote.py

```diff
@@ -412,30 +412,41 @@ def get_many(self, keys):
 class RepositoryCache(RepositoryNoCache):
     """A caching Repository wrapper
 
-    Caches Repository GET operations using a local temporary Repository.
+    Caches Repository GET operations locally.
     """
     def __init__(self, repository):
         super().__init__(repository)
-        tmppath = tempfile.mkdtemp(prefix='borg-tmp')
-        self.caching_repo = Repository(tmppath, create=True, exclusive=True)
-        self.caching_repo.__enter__()  # handled by context manager in base class
+        self.cache = LRUCache(100, dispose=os.unlink)
+
+    def _cache(self, key, data):
+        try:
+            with tempfile.NamedTemporaryFile(prefix='borg-tmp', delete=False) as file:
```

Comment:
usually the temp dir overflows rather quickly, causing all sorts of bad effects.

so, rather use the cache dir?